### ISSUE TITLE
Empty value EC2 tag will fail terraform apply

### ISSUE THREAD

**Issue by cwyl02** [terraform-aws-modules/terraform-aws-autoscaling#102]
Hi guys, 

This is my module definition:

```
module "service-auto-scaling-group" {
  source = "terraform-aws-modules/autoscaling/aws"
  version = "3.0.0"

  ...

  tags = [
   ...
    {
      key = "Reserve"
      value = ""
      propagate_at_launch = true
    }
  ]
}
```

This results in the following tf plan diff:

```
# blah.module.service-auto-scaling-group.aws_autoscaling_group.this[0] will be created
  + resource "aws_autoscaling_group" "this" {
      + ....
      + tags                      = [
          + {
              + "key"                 = "Reserve"
              + "propagate_at_launch" = "true"
            },
        ]
      + ...
    }
```

Notice that the "value" = "" is expected, but it's omitted for some reasons.

And when I run terraform apply, it spit out this error:

```
Error: blah-asg-blah: invalid tag attributes: value missing

 on blah.tf line 60, in resource "aws_autoscaling_group" "this":
  60: resource "aws_autoscaling_group" "this" {
```

09/28 edit:

Expected behavior: 

in the example above,

```
+ resource "aws_autoscaling_group" "this" {
      + ....
      + tags                      = [
          + {
              + "key"                 = "Reserve"
              + "propagate_at_launch" = "true"
              + "value"                          =  ""    // empty value is correctly added
            },
        ]
      + ...
    }
```

Environment: 
terraform 0.12.7
terraform-aws-modules/autoscaling/aws 3.0.0
OS: MacOS 10.13.6

**Comment by cwyl02** [comment]
When I substitute tags to tags_as_map, still got the same behavior

**Comment by bryantbiggs** [comment]
hi @cwyl02 - could you elaborate more on what issue you are looking to be solved? instead of passing an empty string can you just use a value like "none"?

**Comment by cwyl02** [comment]
@bryantbiggs Sorry for the late response. We do have a requirement for this tag having this Reserve tag with empty value - it will be consumed by some AWS billing auditing tool.

**Comment by antonbabenko** [comment]
This has been fixed in #119, so please make sure to use [v3.7.0](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/releases/v3.7.0) which has been just released.
